addons/web_planner/tests/test_web_planner.py

```diff
@@ -26,10 +27,9 @@ def test_00(self):
             }
         )
 
-        self.assertEqual(
-            Planner.prepare_backend_url(self, "web_planner.test_00"),
-            "/web#view_type=list&action=%s" % action.id,
-        )
+        backend_url = Planner.prepare_backend_url(self, "web_planner.test_00")
+        parameters = urls.url_decode(backend_url[backend_url.index('#') + 1:]).to_dict(flat=False)
```

Comment:
Seems fine to me, though I'd suggest using `urls.url_parse(backend_url).fragment` to extract the fragment string, instead of string munging.

Also why the `flat=False` bit`? That seems to only make the test file less readable, none of the tests is using multi-valued keys, why not generate a flat dict?

Reply:
without flat=False the test would not be able to differentiate between "a=b&a=f&c=d" and "a=b&c=d".
for me the flat=False is more like the original string comparison that i replaced.

Reply:
I will update the pull request for url_parse(backen_url).fragment, thanks i didn't have this method on my radar

Reply:
> without flat=False the test would not be able to differentiate between "a=b&a=f&c=d" and "a=b&c=d".

Sure but I don't think we care, I don't think multidicts are used anywhere in Odoo.

Reply:
@xmo-odoo i will remove it if i have too, from my point of view multiple values for parameters can be a thing and "a=b&a=f" is not equal to "a=b". in version 11 at least the ExportDialog in Odoo and account_reports have `<select ...multiple...>` in them which would need to explicitly get all the parameter values if used in a form with action="GET". also using the MultiDict is kind of implicit if one would use request.httprequest.values.getlist(...) as i did in this merge request: https://github.com/odoo/odoo/pull/25323 that would enable/reenable/fix the selecting of multiple variant options in the shop for this url:
http://426059-11-0-c5a3b6.runbot13.odoo.com/shop?search=&attrib=2-3&attrib=2-4&attrib=1-1&attrib=1-2
![screenshot from 2019-01-11 15-40-46](https://user-images.githubusercontent.com/38032588/51040274-4e65be80-15b7-11e9-9e94-e0b809680360.png)

So if i have to i will remove the flat=False, but i would prefer not to.